apps/condo/domains/user/schema/SignInUserService.js

```diff
@@ -0,0 +1,236 @@
+        mutation: 'signInUser',
+        code: BAD_USER_INPUT,
+        type: 'INVALID_PHONE_NUMBER',
+        message: 'Invalid phone number',
+        messageForUser: 'api.user.signInUser.INVALID_PHONE_NUMBER',
+    },
+    USER_NOT_FOUND: {
+        mutation: 'signInUser',
+        code: NOT_FOUND,
+        type: 'USER_NOT_FOUND',
+        message: 'User not found',
+        messageForUser: 'api.user.signInUser.USER_NOT_FOUND',
+    },
+    TOKEN_NOT_FOUND: {
+        mutation: 'signInUser',
+        variable: ['data', 'token'],
+        code: NOT_FOUND,
+        type: 'TOKEN_NOT_FOUND',
+        message: 'Token not found',
+    },
+    MULTIPLE_USERS_FOUND: {
+        mutation: 'signInUser',
+        code: NOT_FOUND,
+        type: 'MULTIPLE_USERS_FOUND',
+        message: 'Multiple users found',
+    },
+    DV_VERSION_MISMATCH: {
+        query: 'signInUser',
+        variable: ['data', 'dv'],
+        code: BAD_USER_INPUT,
+        type: DV_VERSION_MISMATCH,
+        message: 'Wrong value for data version number',
+    },
+    WRONG_SENDER_FORMAT: {
+        query: 'signInUser',
+        variable: ['data', 'sender'],
+        code: BAD_USER_INPUT,
+        type: WRONG_FORMAT,
+        message: 'Invalid format of "sender" field value',
+        correctExample: '{ dv: 1, fingerprint: \'example-fingerprint-alphanumeric-value\'}',
+    },
+}
+
+const getUserPayload = (user, userData) => {
+    const payload = {}
+    for (const [key, value] of Object.entries(userData))  {
+        if (!value) continue
+        if (get(user, key)) continue
+        payload[key] = value
+    }
+    return payload
+}
+
+const checkRequiredUserFields = (requiredFields, userData) => {
+    if (!Array.isArray(requiredFields)) throw new Error('"requiredFields" should be an array')
+
+    const missingFields = []
+    for (const field of requiredFields) {
+        const value = get(userData, field, null)
+        if (!value && typeof value !== 'boolean' && typeof value !== 'number') missingFields.push(field)
+    }
+    return { missingRequiredFields: missingFields.length > 0, missingFields }
+}
+
+const checkDayRequestLimitCounters = async (context, rawIp) => {
+    const ip = rawIp.split(':').pop()
+    const key = `signInUser:ip:${ip}`
+    const byIpCounter = await REDIS_GUARD.incrementDayCounter(key)
+    if (byIpCounter > MAX_ANONYMOUS_REQUESTS_BY_DAY && !IP_WHITE_LIST.includes(ip)) {
+        throw new GQLError(GQL_ERRORS.SMS_FOR_IP_DAY_LIMIT_REACHED, context)
+    }
+}
+
+const SignInUserService = new GQLCustomSchema('SignInUserService', {
+    types: [
+        {
+            access: true,
+            type: 'input SignInUserUserDataInput { phone: String, name: String, email: String, password: String, meta: JSON }',
+        },
+        {
+            access: true,
+            type: 'input SignInUserInput { dv: Int!, sender: SenderFieldInput!, confirmActionToken: ID!, userType: UserTypeType!, userData: SignInUserUserDataInput }',
```

Comment:
`userData` are not required in input?

Reply:
Yes, this is an optional field. For example, when creating a resident, we only need a phone number (from confirm action token). But for an staff, we also need a name and password

So you can pass this data or not.

Reply:
Shold we add a identity field? like `signInUser( { dv, sender, confirmToken, userIdentity?, userType, userData } ) ?

Or we already know the user identity by confirmToken?

Reply:
What do yo think about renaming? may be we should use something like `userIdentityToken`? And specify type like `ConfirmPhoneAction | ConfirmEmailAction` ?

Reply:
> Shold we add a identity field? like `signInUser( { dv, sender, confirmToken, userIdentity?, userType, userData } ) ?
> 
> Or we already know the user identity by confirmToken?

If we have a `confirm action token` -> we know a `phone` -> we can find the `user`

Reply:
> What do yo think about renaming? may be we should use something like `userIdentityToken`? And specify type like `ConfirmPhoneAction | ConfirmEmailAction` ?

Do you mean something like this? 
`{ dv: Int!, sender: SenderFieldInput!, identityToken: ID!, identityTokenType: IdentityTokenTypeType!, userType: UserTypeType!, userData: SignInUserUserDataInput }`
and `IdentityTokenTypeType = ConfirmPhoneAction | ConfirmEmailAction | ...`

Perhaps it really is worth doing this to check exactly a type of token that is specified, and not all of them